### PR TITLE
feat: add support for billing addresses on payment methods

### PR DESCRIPTION
**Pull request by brianium**

### CODE REVIEW

File: tests/Feature/PaymentMethod.php [hunk 1]
```diff
@@ -37,6 +39,28 @@ public function testCreatePaymentMethod()
...
+            ->withEmailAddress('bill.steffen@email.com')
+            ->build();
+
+        $customer = $this->dues->createCustomer($customer);
+        $paymentMethod = new Nonce('fake-valid-mastercard-nonce');
+        $paymentMethod->setCustomer($customer);
+        $paymentMethod->setBillingAddress(new Address(State::Michigan(), '49464'));
+
+        $paymentMethod = $this->dues->createPaymentMethod($paymentMethod);
+        $this->assertInstanceOf(Address::class, $paymentMethod->getBillingAddress());
```
Comment (brianium) [inline tests/Feature/PaymentMethod.php:61]: I imagine our listener will take advantage of this:

```php
$address = $subscription->getCustomer()->getDefaultPaymentMethod()->getBillingAddress();
$calcTax = $this->calcDatTax($address);
```

File: src/Processor/Braintree/Mapper/PaymentMethodMapper.php [hunk 2]
```diff
@@ -37,6 +49,17 @@ public function fromResult($paymentMethod): Token
...
 
+        $billingAddress = $paymentMethod->billingAddress;
+        if ($billingAddress instanceof BraintreeAddress) {
+            $state = null;
+            try {
+                $state = new State($billingAddress->region);
+            } catch (TypeError $e) {
+                // we tried
+            }
+            $token->setBillingAddress(new Address($state, $billingAddress->postalCode));
```
Comment (johncorrelli) [inline src/Processor/Braintree/Mapper/PaymentMethodMapper.php:60]: Does this mean that the state coming back from BT will override anything we set previously? Do I need to set it previously?

File: tests/Model/Address/StateTest.php [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+use PHPUnit\Framework\TestCase;
+use TeamGantt\Dues\Model\Address\State;
+
+final class StateTest extends TestCase
+{
+    public function testValue()
+    {
+        $al = State::Alabama();
+        $this->assertEquals('AL', $al->value);
```
Comment (johncorrelli) [inline tests/Model/Address/StateTest.php:13]:
```diff
- $al = State::Alabama();
- $this->assertEquals('AL', $al->value);
+ $md = State::Maryland();
+ $this->assertEquals('MD', $md->value);
```

:trollface:

**Review by johncorrelli — APPROVED** [review]
Looks good to me! just an implementation question...

File: tests/Feature/PaymentMethod.php [hunk 4]
```diff
@@ -37,6 +39,28 @@ public function testCreatePaymentMethod()
...
+        $customer = (new CustomerBuilder())
+            ->withFirstName('Bill')
+            ->withLastName('Steffen')
+            ->withEmailAddress('bill.steffen@email.com')
+            ->build();
+
+        $customer = $this->dues->createCustomer($customer);
+        $paymentMethod = new Nonce('fake-valid-mastercard-nonce');
+        $paymentMethod->setCustomer($customer);
+        $paymentMethod->setBillingAddress(new Address(State::Michigan(), '49464'));
```
[the remark this answers is not in the document]
Comment (brianium) [reply tests/Feature/PaymentMethod.php:58]: As a member of the corporate media..... he had it coming.

**Pull request merged**